src/ansiblelint/rules/var_naming.md

```diff
@@ -0,0 +1,34 @@
+# var-naming
+
+This rule checks variable names to ensure they conform with requirements.
+
+Variable names must contain only lowercase alphanumeric characters and the underscore `_` character.
+Variable names must also start with either an alphabetic or underscore `_` character.
+
+For more information see the [creating valid variable names](https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#creating-valid-variable-names) topic in Ansible documentation.
+
+## Problematic Code
+
+```yaml
+---
+- name: Example playbook
+  hosts: localhost
+  vars:
+    CamelCase: true # <- Contains a mix of lowercase and uppercase characters.
+    this_is_valid: # <- This is a dict, not a variable
+      ALL_CAPS: bar # <- Contains only uppercase characters.
+      v@r!able: baz # <- Contains special characters.
+```
```

Comment:
That is not correct because variables are only the keys at the first level under vars. They can have any kind of value and if they are dictionaries the keys of these dictionaries are not forced to follow the same rules. Basically only the CameCase example is correct.

Reply:
How about now? I just remove the dictionary to avoid confusion.